src/daemon/normalize.js

```diff
@@ -0,0 +1,12 @@
+const punycode = require('punycode') // eslint-disable-line node/no-deprecated-api
+
+module.exports = function normalize(id) {
+  return id
+    ? punycode
+        .encode(id)
+        .toLowerCase()
+        .replace(/[^a-z0-9-.]/g, '-')
```

Comment:
This ensures that ASCII characters such as spaces, parentheses, commas, and so on, are encoded to a hostname-valid space.